chingu-voyages/v52-tier3-team-38
# Dynamic Landing Pages

# Description
This pull request focuses on enhancing role-based navigation and improving clarity across components by renaming, restructuring, and simplifying pages. The changes are aimed at improving the maintainability and readability of the codebase while ensuring consistent user experiences for guests, admins, and users.

## Related to the following Jira/Trello issue 
- Landing Page

## Main changes explained:
- Add new component:
- - Created GuestHome component by adding a dedicated file at the root and migrating the landing page content.

- Rename components for clarity:
- - Renamed Home to GuestHome for better alignment with its purpose as a guest landing page.
- - Renamed Admin and User profile pages to ensure consistency and clarity in component naming.

- Refactor existing components:
- - Simplified the structure of the GuestHome component by removing unnecessary styles, resulting in cleaner and more efficient code.
- - Updated the Root component to handle user roles more effectively, improving role-based rendering logic.


## How to test:
1. check into Ross/feature/dynamic-landing-pages branch
2. do `npm install` and `npm run dev` to run this PR locally.
3. The landing page should load based on login! 
- If not logged in, should be the normal landing page! 
- If logged in as regular user, it should show regular user profile page! 
- If logged in as admin, it should show admin profile page!


## Screenshots or videos of changes:

![Screenshot 2024-11-28 214705](https://github.com/user-attachments/assets/be1aa209-a1f8-4797-8892-62549f64cc09)
![Screenshot 2024-11-28 214753](https://github.com/user-attachments/assets/51a2d7fb-b8ef-4535-b5a3-e978d5aac282)
![Screenshot 2024-11-28 214825](https://github.com/user-attachments/assets/984d41a2-6022-487c-b87e-13e0c7fd63d7)
![image](https://github.com/user-attachments/assets/6c0bd093-7526-45ea-ad67-7bfc67eb0252)
![image](https://github.com/user-attachments/assets/bf005835-6e26-451e-b9f4-670d49e09dfc)
